src/tdsops.f90

```diff
@@ -84,8 +87,14 @@ function tdsops_init(n, delta, operation, scheme, n_halo, from_to, &
     real(dp), optional, intent(in) :: c_nu, nu0_nu !! params for hypervisc.
 
     integer :: n_stencil
+    integer :: n
+    real(dp) :: delta
+
+    tdsops%mesh = mesh
+    tdsops%dir = dir
+    n = tdsops%mesh%get_n(dir, data_loc)
+    delta = tdsops%mesh%geo%d(dir)
 
-    tdsops%n = n
```

Comment:
A very important variable. I actually think we need a `n_in` and `n_out`. This way it shouldn't be misleading as if its the `n` in `dirps` or somewhere else. Happy to explain this in detail in our meeting on Monday.